lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1197,7 +1197,8 @@ selectCoinsForJoin ctx knownPools getPoolStatus pid wid = do
     curEpoch <- getCurrentEpoch ctx
 
     (utx, action, path) <- withWorkerCtx ctx wid liftE liftE $ \wrk -> do
-        action <- liftHandler
+        -- we never register stake key here, so deposit is irrelevant
+        (action, _) <- liftHandler
```

Comment:
this is a little fishy...

Reply:
> there's no way to trigger a register stake key via this endpoint

I challenge this. When call for the very first time so that this is the first delegation for this wallet, it will result in a stake key registration. I do not understand your rationale here?

Reply:
```hs
    coinSelections :: Server (CoinSelections n)
    coinSelections = (\wid ascd -> case ascd of
        (ApiSelectForPayment ascp) ->
            selectCoins shelley (delegationAddress @n) wid ascp
        (ApiSelectForDelegation (ApiSelectCoinsAction (ApiT action))) ->
            case action of
                Join pid -> selectCoinsForJoin @_ @()
                    shelley
                    (knownPools spl)
                    (getPoolLifeCycleStatus spl)
                    pid
                    (getApiT wid)
                -- if this would be enabled, we'd need to adjust selectCoinsForJoin
                -- to account for deposit
                RegisterKeyAndJoin _ -> throwError err400
                Quit -> selectCoinsForQuit @_ @() shelley wid
        )
```

`RegisterStakeKeyAndJoin` isn't allowed here.

Reply:
Yes, and this is why re-using this type was a bit hacky :upside_down_face: ... This does not translate what's happening in the actual underlying logic. Calls from `selectCoinsForJoin` can totally generate stake key registration with implied key deposits. 

I'd suggest defining a proper api type to reflect this and prevent someone else to be bitten by this. The API action that are supported are either Join/Quit and they map internally to Join / RegisterKeyAndJoin / Quit.

Reply:
I've done:

1. Use a separate Api type for delegation action
2. add `deposits` field to coin selection

```hs
data ApiCoinSelection (n :: NetworkDiscriminant) = ApiCoinSelection
    { inputs :: !(NonEmpty (ApiCoinSelectionInput n))
    , outputs :: ![ApiCoinSelectionOutput n]
    , change :: ![ApiCoinSelectionChange n]
    , certificates :: Maybe (NonEmpty ApiCertificate)
    , deposits :: ![Quantity "lovelace" Natural]
    } deriving (Eq, Generic, Show)
      deriving anyclass NFData
```